### PR TITLE
Renames many things

### PR DESCRIPTION
**Pull request by ashfurrow**
These are meant for the 3.0.0 development branch, since they break things. 

Basically this renames a tonne of things that either no longer make sense, or never did :sweat_smile: Updated the docs and tried to add comments where possible.

### CODE REVIEW

**Comment by ashfurrow** [comment]
These are some pretty big changes to the way we name things – probably best to get input from @Moya/contributors

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,24 @@
...
+- Default class functions have been moved to extensions to prevent inadvertent subclassing.
+- Renamed other identifiers: **Breaking Change**
+  - `MoyaProvider.MoyaEndpointsClosure` to `MoyaProvider.EndpointClosure`
+  - `MoyaProvider.MoyaEndpointResolution` to `MoyaProvider.RequestClosure`
+  - `MoyaProvider.endpointResolver` to `MoyaProvider.requestClosure`
+  - `MoyaProvider.stubBehavior` to `MoyaProvider.stubClosure`
+  - `MoyaProvider` initializer parameter names.
+  - `MoyaCompletion` to `Moya.Completion`
+  - `DefaultEndpointResolution` to `DefaultRequestMapping`
+- Renamed `T` generic types of `MoyaProvider` and `Endpoint` classes to `Target`.
```
Comment (orta) [inline CHANGELOG.md]: :+1:

File: Moya/Endpoint.swift [hunk 2]
```diff
@@ -17,17 +19,23 @@ public enum EndpointSampleResponse {
...
 
     /// Main initializer for Endpoint.
-    public init(URL: String, sampleResponse: EndpointSampleResponse, method: Moya.Method = Moya.Method.GET, parameters: [String: AnyObject] = [String: AnyObject](), parameterEncoding: Moya.ParameterEncoding = .URL, httpHeaderFields: [String: String] = [String: String]()) {
+    public init(URL: String,
+        sampleResponse: EndpointSampleResponse,
+        method: Moya.Method = Moya.Method.GET,
+        parameters: [String: AnyObject]? = nil,
+        parameterEncoding: Moya.ParameterEncoding = .URL,
+        httpHeaderFields: [String: String] = [String: String]()) {
+
```
Comment (orta) [inline Moya/Endpoint.swift]: :+1:

File: Moya/Moya.swift [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
 import Alamofire
 
-/// Block to be executed when a request has completed.
-public typealias MoyaCompletion = (data: NSData?, statusCode: Int?, response: NSURLResponse?, error: ErrorType?) -> ()
-
 /// General-purpose class to store some enums and class funcs.
 public class Moya {
 
+    /// Closure to be executed when a request has completed.
+    public typealias Completion = (data: NSData?, statusCode: Int?, response: NSURLResponse?, error: ErrorType?) -> ()
```
Comment (orta) [inline Moya/Moya.swift]: :+1:

**Comment by orta** [comment]
I definitely prefer the wording here.

**Comment by ashfurrow** [comment]
Cooool, thanks! I'll give some time for others to review, too.

**Comment by jarrodrobins** [comment]
Looks good to me. :+1: 

I particularly like these changes -

     `MoyaProvider.MoyaEndpointResolution` to `MoyaProvider.RequestClosure`
     `MoyaProvider.endpointResolver` to `MoyaProvider.requestClosure`

So much clearer!

**Comment by orta** [comment]
will need a rebase, but good to merge from me

**Comment by aschuch** [comment]
:+1: Good for me :)

**Comment by ashfurrow** [comment]
OK, will merge when I get a chance.

File: Moya/Endpoint.swift (hunk 2 above)
Comment (colinta) [inline Moya/Endpoint.swift]: Parameters is optional, but headers are not, that caught my eye.  Should they be the same?

Comment (colinta) [inline Moya/Endpoint.swift]: Er, should they be *consistent* I meant.

Comment (ashfurrow) [inline Moya/Endpoint.swift]: That's a really great point!

**Comment by ashfurrow** [comment]
OK, so this has had `master` and `3.0.0-develop` merged in. We should be good to go to merge this – once we do, I don't see _too much_ stopping us from releasing 3.0.0 :tada: If anyone wants a final review, now's the time :smile:

**Comment by steam** [comment]
:metal: Nice work. The API is much clearer with these changes.

**Pull request merged**